simulators/main/oeeAPI.py

```diff
@@ -167,21 +167,25 @@ def add_timeslot(self, locationId, timeslot):
         log.warning(f'Cannot create Timeslot for location:{locationId}, content: {response.status_code} - {response.text}, url: {url}, data: {json.dumps(timeslot)}')
         return False
 
-    def get_shiftplan(self, locationId, dateFrom, dateTo):
-        url = f'{self.SHIFTPLAN_REST_ENDPOINT}/{locationId}?dateFrom={dateFrom}&dateTo={dateTo}'
-        response = requests.get(url, headers=C8Y_HEADERS)
+    def get_shiftplan_with_specific_date(self, locationId, dateFrom, dateTo):
+        url = f'{self.SHIFTPLAN_REST_ENDPOINT}/{locationId}'
+        if dateTo and dateFrom:
+            params = {'dateFrom': dateFrom, 'dateTo': dateTo}
+        else:
+            params = None
+        response = requests.get(url, headers=C8Y_HEADERS, params=params)
         if response.ok:
             return response.json()
         log.warning(f'Cannot get shiftplan for {locationId}, url: {url},  response: {response.status_code}: {response.text} ')
         return {'locationId':locationId,'timeslots':{}}
 
-    def get_shiftplan_status(self, locationId,):
```

Comment:
are you sure we dont need the get_shiftplan_status() method anymore?

Reply:
My bad, I mistakenly delete it